interface/patient_file/encounter/delete_form.php

```diff
@@ -77,7 +77,7 @@
 ?>
 <input type="hidden" id="confirm" name="confirm" value="1"/>
 <p>
-<?php echo xlt('You are about to delete the following form from this encounter') . ': ' . text($_GET['formname']); ?>
+<?php echo xlt('You are about to delete the following form from this encounter') . ': ' . xlt(text($_GET['formname'])); ?>
```

Comment:
Hi,

I committed your fix with modification. See below for details (and thanks for the fix!):

Since this is a form title, this is actually a special case (see next comment for details on this).

Usually, you would use xlt() here by itself without the text() function:
``````
<?php echo xlt('You are about to delete the following form from this encounter') . ': ' . xlt($_GET['formname']);

Here a good description of the escaping/translation function:
Sure thing on the different escaping functions used (xl, xlt, attr, text, etc.). There are nice example of them found here:
http://www.open-emr.org/wiki/index.php/Codebase_Security#SQL-Injection_and_Cross-Scripting_Prevention
SEE STEP 4 for examples

For some more background, note these are really shortcut functions to make the code more readable. This is why some of the codebase uses htmlspecialchars (which were done before we had these shortcut functions):
html escaping when not in html attributes:
text = htmlspecialchars($variable,ENT_NOQUOTES)
xlt = htmlspecialchars(xl($variable),ENT_NOQUOTES)

html escaping when in html attributes:
attr = htmlspecialchars($variable,ENT_QUOTES)
xla = htmlspecialchars(xl($variable),ENT_QUOTES)

escaping of javascript literals:
xls = addslashes(xl($variable))

The library for these functions can be found here:
https://github.com/openemr/openemr/blob/master/library/htmlspecialchars.inc.php

hope this helps. if still need clarification on this or other stuff, please always feel free to ask.
(btw, plan to continue on your code review sometime over the next 1-2 days :) )
-brady